app/screens/home/channel_list/categories_list/categories/body/category_body.tsx

```diff
@@ -39,26 +40,40 @@ const CategoryBody = ({sortedChannels, category, hiddenChannelIds, limit, onChan
         return (
             <ChannelItem
                 channel={item}
-                collapsed={category.collapsed}
                 testID={`category.${category.displayName.replace(/ /g, '_').toLocaleLowerCase()}.channel_list_item`}
                 onPress={onChannelSwitch}
             />
         );
-    }, [category.collapsed, onChannelSwitch]);
+    }, [onChannelSwitch]);
+
+    const sharedValue = useSharedValue(category.collapsed);
+
+    useEffect(() => {
+        sharedValue.value = category.collapsed;
+    }, [category.collapsed]);
+
+    const height = ids.length ? ids.length * 40 : 0;
```

Comment:
@shazm there is an issue here where we are not taking into account channels that have been archived and are not the active channel as well as the ones that should not render as they are in the unread section, basically by not taken them into account there is additional space that should not be there.